File: back/src/main/java/com/back/domain/member/dto/MemberDto.java
package com.back.domain.member.dto;

import com.back.domain.member.entity.Member;
import jakarta.validation.constraints.NotNull;

public record MemberDto(
        @NotNull
        Long id,
        @NotNull
        String githubId,
        @NotNull
        String githubLink
) {
    public MemberDto(Member member) {
        this(
                member.getId(),
                member.getGithubId(),
                member.getGithubLink()
        );
    }
}

File: back/src/main/java/com/back/domain/member/dto/MemberDto.kt
package com.back.domain.member.dto

import com.back.domain.member.entity.Member
import jakarta.validation.constraints.NotNull

open class MemberDto(
    @field:NotNull val id: Long,
    val githubId: String?,
    val githubLink: String?
) {
    constructor(member: Member) : this(
        member.id,
        member.githubId,
        member.githubLink
    )
}